### PR TITLE
test: mux and demux

### PR DESCRIPTION
**Pull request by dbogdzix**
Signed-off-by: Daniel Bogdzia <danielx.bogdzia@linux.intel.com>

### CODE REVIEW

File: src/include/sof/audio/mux.h [hunk 1]
```diff
@@ -82,6 +82,51 @@ extern const struct comp_func_map mux_func_map[];
 mux_func mux_get_processing_function(struct comp_dev *dev);
 demux_func demux_get_processing_function(struct comp_dev *dev);
 
+#ifdef UNIT_TEST
+
```
Comment (jajanusz) [inline src/include/sof/audio/mux.h]: remove space here or add one before #endif to have it symmetrical

File: test/cmocka/src/audio/mux/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+SET(arch_src ${PROJECT_SOURCE_DIR}/src/audio/mux/mux.c
+             ${PROJECT_SOURCE_DIR}/src/audio/mux/mux_generic.c)
+
+# make small lib for stripping so we don't have to care
+# about unused missing references
+
+add_compile_options(-fdata-sections -ffunction-sections -DUNIT_TEST)
+link_libraries(-Wl,--gc-sections)
+
+
```
Comment (jajanusz) [inline test/cmocka/src/audio/mux/CMakeLists.txt]: double space

File: src/audio/mux/mux.c [hunk 3]
```diff
@@ -71,7 +72,7 @@ static int mux_set_values(struct comp_data *cd, struct sof_mux_config *cfg)
 	return 0;
 }
 
-static struct comp_dev *mux_new(struct sof_ipc_comp *comp)
+UT_STATIC struct comp_dev *mux_new(struct sof_ipc_comp *comp)
```
Comment (tlauda) [inline src/audio/mux/mux.c]: Do we need to pollute the code like this? We could just use the comp API and call this methods implicitly.

File: src/audio/mux/mux_generic.c [hunk 4]
```diff
@@ -18,9 +19,9 @@
  * \param[in] offset Offset in source buffer.
  * \param[in] mask Routing bitmask for calculating output sample.
  */
-static inline int32_t calc_sample_s16le(struct comp_buffer *source,
-					uint8_t num_ch, uint32_t offset,
-					uint8_t mask)
+UT_STATIC inline int32_t calc_sample_s16le(struct comp_buffer *source,
```
Comment (tlauda) [inline src/audio/mux/mux_generic.c]: Do we need to pollute the code like this? We could just use the comp API and call this methods implicitly.

File: src/audio/mux/mux.c (hunk 3 above)
Comment (jajanusz) [reply src/audio/mux/mux.c]: This is unnecessary. It can be tested via public api.

File: src/audio/mux/mux_generic.c (hunk 4 above)
Comment (jajanusz) [reply src/audio/mux/mux_generic.c]: Defines like this are needed to make it unit-testable. With public API you can test processing as whole (intergration tests), but this is one f.e. is for unit-testing just sample calculation.

**Comment by lgirdwood** [comment]
@dbogdzix what is this for ? Can you describe in PR message and commit messages.

**Comment by jajanusz** [comment]
> @dbogdzix what is this for ? Can you describe in PR message and commit messages.

Updated. I had to finish it instead of original author.

**Comment by jajanusz** [comment]
SOFCI TEST

**Comment by jajanusz** [comment]
SOFCI TEST

**Comment by jajanusz** [comment]
SOFCI TEST

**Pull request merged**